libsolidity/interface/StandardCompiler.cpp

```diff
@@ -901,6 +901,37 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 	if (auto result = checkModelCheckerSettingsKeys(modelCheckerSettings))
 		return *result;
 
+	if (modelCheckerSettings.isMember("contracts"))
+	{
+		auto const& sources = modelCheckerSettings["contracts"];
+		if (!sources.isObject() && !sources.isNull())
+			return formatFatalError("JSONError", "\"contracts\" is not a JSON object.");
+
+		map<string, set<string>> sourceContracts;
+		for (auto const& s: sources.getMemberNames())
+		{
+			if (s.empty())
+				return formatFatalError("JSONError", "source name cannot be empty.");
+
+			auto const& contracts = sources[s];
+			if (!contracts.isArray())
+				return formatFatalError("JSONError", "source contracts must be an array.");
+
+			for (auto const& c: contracts)
+			{
+				if (!c.isString())
+					return formatFatalError("JSONError", "Every contract in settings.modelChecker.contracts must be a string.");
+				if (c.asString().empty())
+					return formatFatalError("JSONError", "contract name cannot be empty.");
+				sourceContracts[s].insert(c.asString());
+			}
+
+			if (sourceContracts[s].empty())
+				return formatFatalError("JSONError", "source contracts must be a non-empty array.");
```

Comment:
The error messages are inconsistent, some start with an upper case letter, other don't. I think we should start with an uppercase letter.

Reply:
Fixed that. I left in lowercase the messages that start with the JSON field name